samplers/aws/xray/internal/client.go

```diff
@@ -139,6 +139,10 @@ func (c *xrayClient) getSamplingRules(ctx context.Context) (*getSamplingRulesOut
 	}
 	defer output.Body.Close()
 
+	if output.StatusCode != http.StatusOK {
```

Comment:
Should this accept any 2xx HTTP responses?

Reply:
TBH I asked myself the same question. As the docs only state a successful response should be 200 OK, I would argue that checking for that exact response code should be perfectly fine.
If you think that should be change, feel free to response, so I will change this :)